File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public String solution(int[] food) {
        
        StringBuilder builder = new StringBuilder();
        
        for (int i = 1; i < food.length; i++) {
            builder.append(Integer.toString(i).repeat(food[i] / 2));
        }
        
        return builder.toString() + "0" + builder.reverse().toString();
    }

}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.js
function solution(food) {
    var answer = '';
    
    for(var i=1; i<food.length; i++){
        if(food[i]%2 != 0){
            food[i] -= 1;
        }
        
        if(food[i]>1){
            for(var j=0; j<food[i]/2; j++){
                answer += i.toString();
            }
        }    
    }
    reverseAnswer = answer.split('').reverse().join('');
    answer = answer + '0' + reverseAnswer;
    

    
    return answer;
}